contracts/src/Gateway.sol

```diff
@@ -509,56 +511,67 @@ contract Gateway is IGatewayBase, IGatewayV1, IGatewayV2, IInitializable, IUpgra
      * APIv2 Internal functions
      */
 
-    // Dispatch all the commands within the batch of commands in the message payload. If a single
-    // command fails, dispatches of subsequent commands are aborted.
+    // Dispatch all the commands within the batch of commands in the message payload. Each command is processed
+    // independently, and failures emits event CommandFailed without stopping execution of subsequent commands.
     function v2_dispatch(InboundMessageV2 calldata message) internal returns (bool) {
+        bool allCommandsSucceeded = true;
+
         for (uint256 i = 0; i < message.commands.length; i++) {
             // check that there is enough gas available to forward to the command handler
             if (gasleft() * 63 / 64 < message.commands[i].gas + DISPATCH_OVERHEAD_GAS_V2) {
                 revert IGatewayV2.InsufficientGasLimit();
             }
+
+            bool commandSucceeded = true;
```

Comment:
This change has introduced some complexity with managing these success flags. I suspect the code be easier to read and maintain if we factor out the the body of the `for` loop into a separate internal`_dispatchCommand(Command calldata command)` function that returns a boolean.

This is just an experiment. It may be worthwhile to try out.

Then the the contents of `_dispatchCommand` can just `return false`. One could say this a more functional style.

Reply:
I tried it - it was a good suggestion - looks simpler. :) Let me know what you think.